Contributing.md

```diff
@@ -0,0 +1,176 @@
+* **List some other game engines / examples where this enhancement exists (_if applicable_).**
+* **Specify the name and version of the OS you're using.**
+
+### Your First Code Contribution
+
+Unsure where to begin contributing to NovelRT? You can start by looking through these `good first issue` and `help wanted` issues:
+
+* [Good first issues](https://github.com/novelrt/NovelRT/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) - issues which should only require a few lines of code, and a test or two.
+* [Help wanted issues](https://github.com/novelrt/NovelRT/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) - issues which should be a bit more involved than `good first issues`.
+
+Both issue lists are sorted by total number of comments. While not perfect, number of comments is a reasonable proxy for impact a given change will have.
+
+#### Local development
+
+For instructions on how to get started with local development of NovelRT, see the following sections in [the engine's README](https://github.com/novelrt/NovelRT#getting-started-with-novelrt).
+
+### Pull Requests
+
+The process described here has several goals:
+
+- Maintain NovelRT's quality
+- Fix problems that are important to users
+- Engage the community in working toward the best possible NovelRT
+- Enable a sustainable system for NovelRT's maintainers to review contributions
+
+Please follow these steps to have your contribution considered by the maintainers:
+
+1. Ensure that the title describes the overall enhancement/fix that is being provided.
+2. The description should summarise the changes being made, and should call out what area(s) of the engine is being affected.
+3. Follow the styleguide as provided by our [`.clang-format` file](https://github.com/novelrt/NovelRT/blob/main/.clang-format). (Our CI will assist you if you are unable to proactively do so, and will provide a git patch to fix most formatting issues, however this is not perfect and should not be guaranteed.)
+4. The pull request that is opened should be [linked to a corresponding issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
+5. Once a pull request is opened, if you have not done so already you will be asked to sign our Contributor License Agreement (CLA).
+5. After you submit your pull request, verify that all [status checks](https://help.github.com/articles/about-status-checks/) are passing <details><summary>What if the status checks are failing?</summary>If a status check is failing, and you believe that the failure is unrelated to your change, please leave a comment on the pull request explaining why you believe the failure is unrelated. A maintainer will re-run the status check for you. If we conclude that the failure was a false positive, then we will open an issue to track that problem with our status check suite.</details>
```

Comment:
We should probably add that the codebase is UK English and ideally spellings should try to be caught before we review it.

Reply:
You caught a mistake with the numbering here too lol